blockchain/upgrade.go

```diff
@@ -65,16 +86,26 @@ func migrateBlockIndex(db database.DB) error {
 
 			var hash chainhash.Hash
 			copy(hash[:], hashBytes[0:chainhash.HashSize])
+			chainContext := blocksMap[hash]
 
-			height, exists := blockHeights[hash]
-			if !exists {
+			if chainContext.height == -1 {
```

Comment:
This will panic if there is no entry for the hash in `blocksMap`.  It probably should either retain the exists check or check it for nil as well.

```Go
if chainContext == nil || chainContext.height == -1 {
...
}
```

Reply:
I feel like this should panic. It should be impossible given that `readBlockTree` iterates over all entries in the bucket and puts them in `blocksMap`.